tests/common/ConfigurationXUnit.cs

```diff
@@ -74,22 +105,64 @@ public AllSupportedPlatformsClassDataAttribute ()
 			dataAttributeType = typeof (T);
 		}
 
-		public override IEnumerable<object []> GetData (MethodInfo testMethod)
+		public override bool SupportsDiscoveryEnumeration () =>
+			!typeof (IDisposable).IsAssignableFrom (dataAttributeType) && !typeof (IAsyncDisposable).IsAssignableFrom (dataAttributeType);
+
+		/// <inheritdoc/>
+		protected override ITheoryDataRow ConvertDataRow (object dataRow)
 		{
-			// we are going to get the instance of the IEnumerable, loop through it and yield the parameters
-			// per platform
-			var enumerable = Activator.CreateInstance (dataAttributeType) as IEnumerable<object []>;
-			if (enumerable is null)
-				yield break;
-			foreach (var platform in Configuration.GetIncludedPlatforms ()) {
-				foreach (var parameters in enumerable) {
-					yield return parameters.Prepend (platform).ToArray ();
+			Guard.ArgumentNotNull (dataRow);
+
+			try {
+				return base.ConvertDataRow (dataRow);
+			} catch (ArgumentException) {
+				throw new ArgumentException (
+					string.Format (
+						CultureInfo.CurrentCulture,
+						"Class '{0}' yielded an item of type '{1}' which is not an 'object?[]', 'Xunit.ITheoryDataRow' or 'System.Runtime.CompilerServices.ITuple'",
+						dataAttributeType.FullName,
+						dataRow?.GetType ().SafeName ()
+					),
+					nameof (dataRow)
+				);
+			}
+		}
+
+		public override async ValueTask<IReadOnlyCollection<ITheoryDataRow>> GetData (
+			MethodInfo testMethod,
+			DisposalTracker disposalTracker)
+		{
+			var classInstance = Activator.CreateInstance (dataAttributeType);
```

Comment:
This would be more AOT-friendly, unless I'm missing why it wouldn't work:

```suggestion
			var classInstance = new T ();
```